app/controllers/navigation/Navigator.scala

```diff
@@ -406,16 +423,36 @@ class Navigator @Inject()(
       routes.CommodityDetailsController.displayPage(mode, itemId)
 
   private def additionalDocumentsSummaryPreviousPage(cacheModel: ExportsDeclaration, mode: Mode, itemId: String): Call =
-    if (cacheModel.itemBy(itemId).flatMap(_.additionalInformation).exists(_.items.nonEmpty))
+    routes.IsLicenseRequiredController.displayPage(mode, itemId)
+
+  private def additionalDocumentsPreviousPage(cacheModel: ExportsDeclaration, mode: Mode, itemId: String): Call = {
+
+    val isLicenseRequired = cacheModel.itemBy(itemId).exists(_.isLicenseRequired.contains(true))
+
+    if (cacheModel.isAuthCodeRequiringAdditionalDocuments || isLicenseRequired) routes.IsLicenseRequiredController.displayPage(mode, itemId)
+    else routes.AdditionalDocumentsRequiredController.displayPage(mode, itemId)
+
+  }
+
+  private def additionalDocumentsSummaryClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode, itemId: String): Call =
+    if (cacheModel.listOfAdditionalInformationOfItem(itemId).nonEmpty)
       routes.AdditionalInformationController.displayPage(mode, itemId)
     else
       routes.AdditionalInformationRequiredController.displayPage(mode, itemId)
 
-  private def additionalDocumentsPreviousPage(cacheModel: ExportsDeclaration, mode: Mode, itemId: String): Call =
+  private def isLicenseRequiredPreviousPage(cacheModel: ExportsDeclaration, mode: Mode, itemId: String): Call =
+    if (cacheModel.listOfAdditionalInformationOfItem(itemId).nonEmpty)
+      routes.AdditionalInformationController.displayPage(mode, itemId)
+    else
+      routes.AdditionalInformationRequiredController.displayPage(mode, itemId)
+
+  private def additionalDocumentsClearancePreviousPage(cacheModel: ExportsDeclaration, mode: Mode, itemId: String): Call =
     if (cacheModel.listOfAdditionalDocuments(itemId).nonEmpty)
       routes.AdditionalDocumentsController.displayPage(mode, itemId)
```

Comment:
Not your logic, but I don't get this behaviour - if there are no documents it displays /add-additional-documentation again? :confused:

Reply:
If there are documents it shows the list